hasher-matcher-actioner/webapp/src/pages/Matches.jsx

```diff
@@ -0,0 +1,278 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+ */
+
+import React, {useState, useEffect} from 'react';
+import {useHistory} from 'react-router-dom';
+import {PropTypes} from 'prop-types';
+import {
+  Container,
+  Col,
+  Button,
+  Row,
+  Collapse,
+  InputGroup,
+  Form,
+  FormControl,
+} from 'react-bootstrap';
+import Spinner from 'react-bootstrap/Spinner';
+import classNames from 'classnames';
+
+import {
+  fetchAllMatches,
+  fetchMatchesFromContent,
+  fetchMatchesFromSignal,
+} from '../Api';
+import {timeAgo} from '../utils/DateTimeUtils';
+import ContentMatchPane from '../components/ContentMatchPane';
+import {useQuery} from '../utils/QueryParams';
+
+export default function Matches() {
+  const query = useQuery();
+  const searchQuery = query.get('contentId') || query.get('signalId');
+  let searchInAttribute;
```

Comment:
not blocking: the state management of this variable confused me a little because of the ways it can be updated and assigned to local but can't think of a cleaner way at first glance.